internal/server/actions.go

```diff
@@ -0,0 +1,172 @@
+package server
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/slack-go/slack"
+	"github.com/writhe/kudosaurus/internal/misc"
+)
+
+// InteractivityPayload - workaround for missing nlopes/slack block types
+type InteractivityPayload struct {
+	Type    string `json:"block_actions"`
+	Actions []struct {
+		ActionID      string   `json:"action_id"`
+		SelectedUsers []string `json:"selected_users"`
+	} `json:"actions"`
+}
+
+func (s *CommandServer) handleDeleteKudo(
+	id string,
+	slackUser slack.User,
+	res http.ResponseWriter,
+) {
+	k, err := s.store.GetKudo(id)
+
+	if err != nil {
+		s.closeWithError(err, http.StatusNotFound, res)
+		return
+	}
+
+	if k.PersonFrom.ID != slackUser.ID {
+		s.closeWithError(errors.New("this is not your kudo to delete"), http.StatusUnauthorized, res)
+		return
+	}
+
+	if err := s.store.RemoveKudo(id); err != nil {
+		s.closeWithError(err, http.StatusInternalServerError, res)
+		return
+	}
+
+	res.WriteHeader(http.StatusOK)
```

Comment:
WriteHeader is really only useful for setting any status besides 200 OK, as this one is assigned by default.
So unless you changed it on any earlier stage, these `res.WriteHeader(http.StatusOK)` can be removed

Reply:
Really? I thought I need an explicit call to `WriteHeader` or `Write` to actually - you know - write a response. Note that what happens below this line is a separate request, not a response.

Reply:
`// WriteHeader sends an HTTP response header with the provided
	// status code.
	//
	// If WriteHeader is not called explicitly, the first call to Write
	// will trigger an implicit WriteHeader(http.StatusOK).
	// Thus explicit calls to WriteHeader are mainly used to
	// send error codes.
	//
	// The provided code must be a valid HTTP 1xx-5xx status code.
	// Only one header may be written. Go does not currently
	// support sending user-defined 1xx informational headers,
	// with the exception of 100-continue response header that the
	// Server sends automatically when the Request.Body is read.
	WriteHeader(statusCode int)`

OK, now I see that you don't `Write` anywhere, so you actually need to  `WriteHeader`